File: benchmarks/fixtures/go/semantic-cases/greeter/runner.go
package greeter

import (
	"fmt"

	util "example.com/squeezy/go-semantic-cases/util"
)

const DefaultName = "Ada"

var Shared = Runner{}

type Greeter interface {
	Greet(name string) string
}

type Runner struct {
	Name string
	Greeter
}

func NewRunner(name string) Runner {
	return Runner{Name: name}
}

func (r Runner) Greet(name string) string {
	fmt.Println(name)
	helper()
	return util.Format(name)
}

func helper() string {
	return DefaultName
}

File: benchmarks/fixtures/scala/semantic-cases/src/main/scala/example/app/Runner.scala
package example.app

import example.util.*
import example.services.{Greeter, FriendlyGreeter}
import example.ext.*

class Runner(val greeter: Greeter) {
  def run(): String = {
    val name = defaultName.toString
    val greeting = greeter.greet(name)
    "hello".shout
    greeting
  }
}

def buildDefault(): Runner = new Runner(Greeter.default)
